File: defenestratio-impl/backend/src/main/java/com/bsuir/defenestratio/entity/UserChallenge.java
package com.bsuir.defenestratio.entity;

import lombok.Data;

import javax.persistence.*;
import java.sql.Date;

@Data
@Entity
@Table(name = "user_challenges")
public class UserChallenge {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "status")
    @Enumerated(EnumType.STRING)
    private ChallengeStatus status;

    @Column(name = "due_date")
    private Date dueDate;

    @Column(name = "user_id")
    private Long userId;

    @ManyToOne
    @JoinColumn(name = "challenge_id", referencedColumnName = "id")
    private Challenge challenge;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "result_id", referencedColumnName = "id")
    private ChallengeResult challengeResult;
}

File: defenestratio-impl/frontend/src/app/models/user.challenge.ts
import {ChallengeResult} from "./challenge.result";
import {ChallengeModel} from "./challenge.model";

export class UserChallenge {
  id: string;
  status: string;
  dueDate: Date;
  userId: string;
  challenge: ChallengeModel;
  challengeResult: ChallengeResult
}
